streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -623,7 +618,11 @@ private long sumOfChangelogOffsets(final TaskId id, final Map<TopicPartition, Lo
     }
 
     private void closeTaskDirty(final Task task) {
-        task.prepareCloseDirty();
+        try {
+            task.suspend();
+        } catch (final RuntimeException swallow) {
```

Comment:
I could get a follow-up newbie ticket, but it seems that we have a couple of catch and swallow cases in the task manager with `clean` flag, does it make sense to extract the `executeAndMaybeSwallow` to TaskManager class and share between cases?

Reply:
SGTM. Can you create a ticket?